test/foundry/FulfillBasicOrderTest.t.sol

```diff
@@ -155,110 +153,32 @@ contract FulfillBasicOrderTest is BaseOrderTest, LowLevelHelpers {
-            fulfillBasicOrderSignature,
+        // Get the calldata that will be passed into fulfillOrder.
+        bytes memory fulfillOrderCalldata = abi.encodeWithSelector(
+            consideration.fulfillBasicOrder.selector,
             _basicOrderParameters
         );
 
-        if (overwriteTotalRecipientsLength) {
-            // Get the additional recipients length from the calldata and
-            // store the length - amountToSubtractFromTotalRecipients in the calldata
-            // so that the length value does _not_ accurately represent the actual
-            // total recipients length.
-            assembly {
-                let additionalRecipientsLengthOffset := add(
-                    fulfillBasicOrderCalldata,
-                    0x264
-                )
-                let additionalRecipientsLength := mload(
-                    additionalRecipientsLengthOffset
-                )
-                mstore(
-                    additionalRecipientsLengthOffset,
-                    sub(
-                        additionalRecipientsLength,
-                        amountToSubtractFromTotalRecipients
-                    )
-                )
-            }
-        }
-
-        address considerationAddress = address(consideration);
-        uint256 calldataLength = fulfillBasicOrderCalldata.length;
-        bool success;
-
-        assembly {
-            // Call fulfillBasicOrders
-            success := call(
-                gas(),
-                considerationAddress,
-                0,
-                // The fn signature and calldata starts after the
-                // first OneWord bytes, as those initial bytes just
-                // contain the length of fulfillBasicOrderCalldata
-                add(fulfillBasicOrderCalldata, OneWord),
-                calldataLength,
-                // Store output at empty storage location,
-                // identified using "free memory pointer".
-                mload(0x40),
-                OneWord
-            )
-        }
-
-        // If overwriteTotalRecipientsLength is True, the call should
-        // have failed (success should be False) and if overwriteTotalRecipientsLength is False,
-        // the call should have succeeded (success should be True).
-        assertEq(!overwriteTotalRecipientsLength, success);
-
-        if (overwriteTotalRecipientsLength) {
-            // Expect a revert if the additional recipients length is too small (e.g. 1 was subtracted).
-            vm.expectRevert();
-        }
-
-        if (!success) {
-            // Revert and pass the revert reason along if one was returned.
-            _revertWithReasonIfOneIsReturned();
-
-            // Otherwise, revert with a generic error message.
-            revert();
-        }
+        _performTestFulfillOrderRevertInvalidArrayLength(
+            consideration,
+            myOrder,
+            fulfillOrderCalldata,
+            0x44,
```

Comment:
I believe the offset of order parameters in the calldata appears to be at 0x24. However, both here and for FulfillOrderTest.t.sol, I had to use the actual offset of order parameters + 0x20 in order to get (a) the right output in my debug logs (e.g. `emit log_named_bytes32(...)`) (b) get the tests to pass

I'm guessing that's because the length of the calldata is 0x20, and that length field would be stored at the start of the calldata array but would not be printed.

@0age does this sound reasonable?

Reply:
I added this comment, ideally we could clarify this `and 0x20 bytes until the start of order parameters` further but I don't have a good explanation

Reply:
first there's the function selector (0x00-0x04); then first word (0x04-0x24) is the offset of the OrderParameters struct; next word (0x24-0x44) is the fulfillerConduitKey; _then_ you've got the data for the order parameters at 0x44 in calldata

Reply:
I'm confused, where does the fulfiller's conduit key get included here?
```
bytes memory fulfillOrderCalldata = abi.encodeWithSelector(
            consideration.fulfillBasicOrder.selector,
            _basicOrderParameters
        );
```

Reply:
Resolved offline